ompi/mca/coll/adapt/coll_adapt_ibcast.c

```diff
@@ -335,7 +336,7 @@ int ompi_coll_adapt_ibcast(void *buff, int count, struct ompi_datatype_t *dataty
     }
 
     return ompi_coll_adapt_ibcast_generic(buff, count, datatype, root, comm, request, module,
-                                          adapt_module_cached_topology(module, comm, root, mca_coll_adapt_component.adapt_ibcast_algorithm),
+                                          ompi_mca_coll_adapt_module_cached_topology(module, comm, root, mca_coll_adapt_component.adapt_ibcast_algorithm),
```

Comment:
Right now it seems like `ompi_coll_*` is used as a common prefix in the collective components. I didn't find any symbol named `ompi_mca_coll_*`. We should stick to the naming scheme we have.

This is probably true for other namespaces as well (`ompi_common` instead of `ompi_mca_common`)

Reply:
I was trying to follow what's described in developers/source-code.rst , which specifies the symbol naming rules:

  * Public symbols must be prefixed in components with
    ``<project>_<framework>_<component>``, where ``<project>`` is one
    of ``mca``, ``ompi``, ``oshmem``, or ``opal``.  Note that `mca`
    used to be the most common, but it has fallen out of favor
    compared to the other ``<project>`` prefixes.  When in doubt about
    whether a symbol is public, be safe and add the prefix.
  * Non-public symbols must be declared ``static`` or otherwise made to
    not appear in the global scope.

After discussion with @jjhursey, I understood this required naming symbols as I have, where the '_mca_' is required since the source file is in the ompi/mca/... source directory.. I'll follow whatever naming convention is wanted, just need to know which way to do it.

Reply:
I'd say the `<project>` in this case is `ompi`, the `<framework>` is `coll` and  the `<component>` is `adapt`. I'm not sure if there was a `mca` project in the past. We might want to update the docs.